### ISSUE TITLE
After initial set-up, failure to validate podcast files; "file not found" for "filesize" under Podcast Episode

### ISSUE THREAD

**Issue by accozzaglia** [podlove/podlove-publisher#690]
Perhaps this is resolved elsewhere, but I have yet to locate another instance of it.

In setting up Podlove for the first time, I have failed, despite hours of troubleshooting the root of the issue. While I'm not a developer, I'm also not new to working with back-end site configurations, including WP. 

This test podcast's episode assets are in two parts: an M4A audio feed and a jpg image.

In this case: 

1) The slug for the podcast is labelled "2355".

2) The podcast media upload location is 'http://accozzaglia.ca/wp-content/podcasts/'

3) Under this naming convention, I named the files relevant to the first podcast as 2355001.\*:
![screen shot 2015-03-07 at 13 16 57a](https://cloud.githubusercontent.com/assets/11367174/6542630/84442310-c4cc-11e4-9d26-86e293d356dc.jpg)

4) That said, despite the podcast new post's slug labelled as '2355001', there is still a failure to verify/locate podcast media files:
![screen shot 2015-03-07 at 13 20 00](https://cloud.githubusercontent.com/assets/11367174/6542645/0b1ca84e-c4cd-11e4-892f-038909dc1039.jpg)
![screen shot 2015-03-07 at 13 23 47](https://cloud.githubusercontent.com/assets/11367174/6542657/58e3c210-c4cd-11e4-8732-0a59238a016c.jpg)

5) Despite repeating the setup process from scratch (for what I believe is now the fourth time), all ending with errors (including the log indicating that the filename gets truncated to '235500.\*', not the designated '2355001.\*', followed by further truncations of '23550.\*' and '2355.\*') — and now, the latest, an issue with the Cloudflare IPv6 address which handles CDN for my site — I'm at an utter loss. There is either something impressively simple which I have overlooked, or else there is something fundamentally incompatible with the podlove plug-in and my site's basic configuration.

The overview would not suggest this.

`Website             http://accozzaglia.ca`
`PHP Version         5.4.34`
`WordPress Version   3.6.1`
`Publisher Version   2.0.5`
`Web Player Version  2.0.17`
`curl Version        7.24.0`
`iconv               available`
`simplexml           ok`
`max_execution_time  300`
`upload_max_filesize 50M`
`memory_limit        256M`
`disable_classes     `
`disable_functions   `
`permalinks          ok`
`podcast_settings    `
`web_player          ok`
`podlove_cache       on`
`assets              `
`	m4a    audio/mpeg`
`	jpg    image/jpeg`
` `
`0 errors`
`0 notices`
`Nice, Everything looks fine!`

6) Remedial efforts, such as the "Attempt Repair" button, have not changed the outcome of the above.

7) An excerpt from the dashboard log. This is typical of what repeats when running through the same motions:

[2015-03-07 09:34:00] Curl Error: Failed to connect to 2400:cb00:2048:1::681c:1d26: Network is unreachable
[2015-03-07 09:34:00] Unexpected http response when trying to access remote media file. HTTP Status: 404
[2015-03-07 09:34:00] --- # Can't reach http://accozzaglia.ca/wp-content/podcasts/235500.jpg --- # Please include this output when you report a bug url: http://accozzaglia.ca/wp-content/podcasts/235500.jpg content_type: http_code: 404 header_size: 0 request_size: 218 filetime: -1 ssl_verify_result: 0 redirect_count: 0 total_time: 0.068307 namelookup_time: 0.023739 connect_time: 0.045252 pretransfer_time: 0.045355 size_upload: 0 size_download: 0 speed_download: 0 speed_upload: 0 download_content_length: -1 upload_content_length: 0 starttransfer_time: 0.068209 redirect_time: 0 certinfo: Array ( ) primary_ip: 104.28.28.38 primary_port: 80 local_ip: 69.89.31.150 local_port: 54661 redirect_url: php_open_basedir: php_safe_mode: php_curl: 1 curl_exec: 1
[2015-03-07 09:34:04] Curl Error: The requested URL returned error: 404
[2015-03-07 09:34:04] Unexpected http response when trying to access remote media file. HTTP Status: 404
[2015-03-07 09:34:04] --- # Can't reach http://accozzaglia.ca/wp-content/podcasts/23550.m4a --- # Please include this output when you report a bug url: http://accozzaglia.ca/wp-content/podcasts/23550.m4a content_type: http_code: 404 header_size: 0 request_size: 217 filetime: -1 ssl_verify_result: 0 redirect_count: 0 total_time: 0.316735 namelookup_time: 0.103583 connect_time: 0.124774 pretransfer_time: 0.124903 size_upload: 0 size_download: 0 speed_download: 0 speed_upload: 0 download_content_length: -1 upload_content_length: 0 starttransfer_time: 0.316677 redirect_time: 0 certinfo: Array ( ) primary_ip: 104.28.28.38 primary_port: 80 local_ip: 69.89.31.150 local_port: 54673 redirect_url: php_open_basedir: php_safe_mode: php_curl: 1 curl_exec: 1
[2015-03-07 09:34:06] Curl Error: The requested URL returned error: 404
[2015-03-07 09:34:06] Unexpected http response when trying to access remote media file. HTTP Status: 404
[2015-03-07 09:34:06] --- # Can't reach http://accozzaglia.ca/wp-content/podcasts/23550.jpg --- # Please include this output when you report a bug url: http://accozzaglia.ca/wp-content/podcasts/23550.jpg content_type: http_code: 404 header_size: 0 request_size: 217 filetime: -1 ssl_verify_result: 0 redirect_count: 0 total_time: 0.227694 namelookup_time: 0.023338 connect_time: 0.044434 pretransfer_time: 0.044527 size_upload: 0 size_download: 0 speed_download: 0 speed_upload: 0 download_content_length: -1 upload_content_length: 0 starttransfer_time: 0.227634 redirect_time: 0 certinfo: Array ( ) primary_ip: 104.28.29.38 primary_port: 80 local_ip: 69.89.31.150 local_port: 33573 redirect_url: php_open_basedir: php_safe_mode: php_curl: 1 curl_exec: 1
[2015-03-07 09:34:08] Curl Error: The requested URL returned error: 404
[2015-03-07 09:34:08] Unexpected http response when trying to access remote media file. HTTP Status: 404
[2015-03-07 09:34:08] --- # Can't reach http://accozzaglia.ca/wp-content/podcasts/2355.jpg --- # Please include this output when you report a bug url: http://accozzaglia.ca/wp-content/podcasts/2355.jpg content_type: http_code: 404 header_size: 0 request_size: 216 filetime: -1 ssl_verify_result: 0 redirect_count: 0 total_time: 0.078203 namelookup_time: 0.028449 connect_time: 0.049625 pretransfer_time: 0.049735 size_upload: 0 size_download: 0 speed_download: 0 speed_upload: 0 download_content_length: -1 upload_content_length: 0 starttransfer_time: 0.078125 redirect_time: 0 certinfo: Array ( ) primary_ip: 104.28.29.38 primary_port: 80 local_ip: 69.89.31.150 local_port: 33577 redirect_url: php_open_basedir: php_safe_mode: php_curl: 1 curl_exec: 1
[2015-03-07 09:34:08] Curl Error: Failed to connect to 2400:cb00:2048:1::681c:1d26: Network is unreachable
[2015-03-07 09:34:08] Unexpected http response when trying to access remote media file. HTTP Status: 404
[2015-03-07 09:34:08] --- # Can't reach http://accozzaglia.ca/wp-content/podcasts/2355.m4a --- # Please include this output when you report a bug url: http://accozzaglia.ca/wp-content/podcasts/2355.m4a content_type: http_code: 404 header_size: 0 request_size: 216 filetime: -1 ssl_verify_result: 0 redirect_count: 0 total_time: 0.235264 namelookup_time: 0.001112 connect_time: 0.02237 pretransfer_time: 0.022457 size_upload: 0 size_download: 0 speed_download: 0 speed_upload: 0 download_content_length: -1 upload_content_length: 0 starttransfer_time: 0.235176 redirect_time: 0 certinfo: Array ( ) primary_ip: 104.28.28.38 primary_port: 80 local_ip: 69.89.31.150 local_port: 54682 redirect_url: php_open_basedir: php_safe_mode: php_curl: 1 curl_exec: 1

* * * * *

I apologize if this is insufficient data to work with or if I missed a similar issue resolved elsewhere previously. Thanks.

**Comment by eteubert** [comment]
I see two issues, the main one:  your cloudflare doesn't respond with a `Content-Length` Header, so the Publisher has no way of finding out the size of your file — and fail. I reproduced your setup and found that in this case, there are no log entries. So I _assume_ that next to the wrong calls you posted, the correct calls are made, but failed silently. Just fix your CDN :)

While you are at it, it should not return `audio/mpeg` as mime-type for `m4a`, but `audio/mp4`. I see you edited the file types, but you should really fix the CDN instead.

I hope this helps. Please report back if you have further questions.

**Comment by accozzaglia** [comment]
The main issue was resolved by creating a CNAME alias to bypass the CDN entirely. This remedied that issue. The other issue (on mime types) I've resigned to Podlove hating the encoding of my m4a file. It doesn't impact playback.

Unfortunately, we're onto a new issue. I look forward to gushing glowingly over podlove being awesome, but the obstacles thus far have bordered on vexing. More in the next issue.
